solutions/cf-585D.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: cf-585D
+source: CF
+title: Lizard Era: Beginning
+author: Albert Ye
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/20898)
+
+In sum, we notice that the size of the array is 25. A naive brute-force would take $\mathcal O(3^n)$ which is too slow. But, a meet-in-the-middle solution would take at most $\mathcal O(\sqrt{3^n} \log (\sqrt{3^n}))$, which is a significant optimization. 
```

Comment:
Also why for the second sentence? prove it!

Reply:
agreed!